modules/dbnd-airflow/src/dbnd_airflow/scheduler/single_dag_run_job.py

```diff
@@ -687,9 +687,8 @@ def _execute(self, session=None):
 
         ti_status.total_runs = 1  # total dag runs in backfill
 
+        dag_run = self._get_dag_run(session=session)
```

Comment:
_get_dag_run can move dag into running state

Reply:
I would move it into try so finally will handle
we can extract logic into _validate_dag_run_complet_state function